sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -0,0 +1,318 @@
+     */
+    private static Mono<okio.ByteString> aggregate(Flux<ByteBuffer> bbFlux) {
+        Objects.requireNonNull(bbFlux);
+        return Mono.using(okio.Buffer::new,
+                buffer -> bbFlux.reduce(buffer, (b, byteBuffer) -> {
+                    try {
+                        b.write(byteBuffer);
+                        return b;
+                    } catch (IOException ioe) {
+                        throw Exceptions.propagate(ioe);
+                    }
+                })
+                .map(b -> okio.ByteString.of(b.readByteArray())),
+                okio.Buffer::clear)
+            .switchIfEmpty(EMPTY_BYTE_STRING_MONO);
+    }
+
+    /**
+     * An implementation of azure-core HttpResponse for OkHttp.
+     */
+    private static class OkHttpResponse extends HttpResponse {
+        private final okhttp3.Response inner;
+        private final HttpHeaders headers;
+        private final static int BYTE_BUFFER_CHUNK_SIZE = 1024;
+
+        public OkHttpResponse(okhttp3.Response inner, HttpRequest request) {
+            this.inner = inner;
+            this.headers = fromOkHttpHeaders(this.inner.headers());
+            super.request(request);
+        }
+
+        @Override
+        public int statusCode() {
+            return this.inner.code();
+        }
+
+        @Override
+        public String headerValue(String name) {
+            return this.headers.value(name);
+        }
+
+        @Override
+        public HttpHeaders headers() {
+            return this.headers;
+        }
+
+        @Override
+        public Flux<ByteBuffer> body() {
+            return this.responseBody() != null
+                    ? toFluxByteBuffer(this.responseBody().byteStream())
+                    : Flux.empty();
+        }
+
+        @Override
+        public Mono<byte[]> bodyAsByteArray() {
+            if (this.responseBody() == null) {
+                return Mono.empty();
+            } else {
+                return Mono.using(() -> this.responseBody(),
+                        rb -> {
+                            try {
+                                byte[] content = rb.bytes();
+                                return content.length == 0 ? Mono.empty() : Mono.just(content);
+                            } catch (IOException ioe) {
+                                throw Exceptions.propagate(ioe);
+                            }
+                        },
+                        rb -> rb.close());
+            }
+        }
+
+        @Override
+        public Mono<String> bodyAsString() {
+            if (this.responseBody() == null) {
+                return Mono.empty();
+            } else {
+                return Mono.using(() -> this.responseBody(),
+                        rb -> {
+                            try {
+                                String content = rb.string();
+                                return content.length() == 0 ? Mono.empty() : Mono.just(content);
```

Comment:
Can this instead be `return Mono.just(content)`? Any reason to return `Mono.empty()` instead of an empty string?

Reply:
see below comment